### DISCUSSION TITLE
Re-assign Drag Snapping Configurations

### DISCUSSION THREAD

**Discussion by plasmatech8** [rxhanson/Rectangle#775] [Ideas]
I was thinking it would be nice to be able to re-assign the drag areas (and/or create custom snap areas for each area).

I have a super-ultrawide monitor. I like using window dragging to the bottom of the screen to snap to thirds - however the middle third is just a bit too small for me. 

I was thinking that it would be nice to be able to:
* Re-assign bottom-snapping to 25/50/25 snap areas instead of thirds
  * (or modify existing thirds to custom sizes like 30/40/30)
* Re-assign corner-snapping to [12.5]/50/25 eighths instead of quarters
  * (or modify existing quarters to custom sizes like [15]/40/30)

Is this possible?

**Comment by plasmatech8** [comment]
also was thinking about re-assigning top-snap to center-half instead of maximize, because it is a bit wide on a super ultra-wide.

**Comment by rxhanson** [comment]
There is not a way to do this at the moment. Ideally, this would be something configurable in the UI, which is kind of a large task and why I haven't gotten around to it. If anyone wants to give it a go, I have no reason to turn down a good PR for it.

**Comment by marauder37** [comment]
I saw on another snapping app that their suggestion for UW monitors is "thirds, with a somewhat bigger middle third". I came looking for this feature because it's a good solution. On a UW, equal thirds is a bit too small for a typical "do stuff in the middle third, observe result in right third, reference material or terminal window in left third" arrangement. But any arrangement not based on a 3x1 grid is wrong for UW, having the main thing be the middle thing is what makes UW great.

I'm sorry I can't recall which snap app came up with "slightly bigger middle third", it may even have been one of the Windows ones. 

Is there a place I would start for implementing this?

**Reply by rxhanson** [reply]
Start with `CompoundSnapArea` where you can see the existing snap areas that have multiple areas for an edge.
